R/get_coding_ssm_status.R

```diff
@@ -0,0 +1,191 @@
+#' This function also has a lot of filtering and convenience parameters giving the user full control of the return. For more information, refer to the parameter descriptions and examples.
+#' Is this function not what you are looking for? Try one of the following, similar, functions; [GAMBLR::get_coding_ssm], [GAMBLR::get_ssm_by_patients], [GAMBLR::get_ssm_by_sample],
+#' [GAMBLR::get_ssm_by_samples], [GAMBLR::get_ssm_by_region], [GAMBLR::get_ssm_by_regions]
+#'
+#' @param gene_symbols A vector of gene symbols for which the mutation status will be tabulated. If not provided, lymphoma genes will be returned by default.
+#' @param these_samples_metadata The metadata for samples of interest to be included in the returned matrix. Only the column "sample_id" is required. If not provided, the matrix is tabulated for all available samples as default.
+#' @param from_flatfile Optional argument whether to use database or flat file to retrieve mutations. Default is TRUE.
+#' @param augmented default: TRUE. Set to FALSE if you instead want the original MAF from each sample for multi-sample patients instead of the augmented MAF.
+#' @param min_read_support Only returns variants with at least this many reads in t_alt_count (for cleaning up augmented MAFs).
+#' @param maf_path If the status of coding SSM should be tabulated from a custom maf file, provide path to the maf in this argument. The default is set to NULL.
+#' @param maf_data Either a maf loaded from disk or from the database using a get_ssm function.
+#' @param include_hotspots Logical parameter indicating whether hotspots object should also be tabulated. Default is TRUE.
+#' @param keep_multihit_hotspot Logical parameter indicating whether to keep the gene annotation as mutated when the gene has both hot spot and non-hotspot mutation. Default is FALSE. If set to TRUE, will report the number of non-hotspot mutations instead of tabulating for just mutation presence.
+#' @param recurrence_min Integer value indicating minimal recurrence level.
+#' @param seq_type The seq_type you want back, default is genome.
+#' @param projection Specify projection (grch37 or hg38) of mutations. Default is grch37.
+#' @param review_hotspots Logical parameter indicating whether hotspots object should be reviewed to include functionally relevant mutations or rare lymphoma-related genes. Default is TRUE.
+#' @param genes_of_interest A vector of genes for hotspot review. Currently only FOXO1, MYD88, and CREBBP are supported.
+#' @param genome_build Reference genome build for the coordinates in the MAF file. The default is hg19 genome build.
+#' @param include_silent Logical parameter indicating whether to include silent mutations into coding mutations. Default is TRUE.
+#'
+#' @return A data frame with tabulated mutation status.
+#'
+#' @import dplyr tidyr
+#' @export
+#'
+#' @examples
+#' coding_tabulated_df = get_coding_ssm_status(maf_data = grande_maf,
```

Comment:
Since the Grande maf is now deprecated in GAMBLR.data, this example should be updated to not use this resource.